### DISCUSSION TITLE
use go embed for templates

### DISCUSSION THREAD

**Discussion by wangzuo** [99designs/gqlgen#1915] [Ideas]
go embed makes it's easy to work with template files in go. I have made a small update to add embed support in templates on [my fork](https://github.com/swiftcarrot/gqlgen/blob/master/codegen/templates/templates.go#L83). I can prepare a pull request if it works for everyone. Any thought on this?

**Comment by haswalt** [comment]
This will only work for people on go 1.16

**Comment by sagikazarmark** [comment]
Now that 1.17 is out, I don't think that's such a bad thing.

**Comment by sagikazarmark** [comment]
Looking at the code, I think there is a way to implement it in a backwards compatible way: when available, load the template from an embedded source and use the `Template` parameter in template options. This would put some complexity on the client/plugin side, but could later be refactored when pre-1.16 support is dropped.

On the other hand, we are getting closer to Go 1.18, so I don't know if supporting versions earlier than Go 1.16 is important. That requires a maintainer decision (@vektah ?).

Last, but not least: using embedding would solve #910.

I'd be happy to work on the solution I proposed above if we can agree on the details.
